infrastructure/code/bundle.go

```diff
@@ -317,26 +320,68 @@ func (b *Bundle) autofixFunc(ctx context.Context, issue snyk.Issue) func() *snyk
 				b.notifier.SendShowMessage(sglsp.MTError, "Something went wrong. Please try again. Request ID: "+b.requestId)
 				return nil
 			case <-pollingTicker.C:
-				edit, complete := pollFunc()
+				fix, complete := pollFunc()
 				if !complete {
 					continue
 				}
-				if edit != nil {
-					b.notifier.SendShowMessage(sglsp.Info, "Congratulations! 🎉 You’ve just fixed this "+issueTitle(issue)+" issue.")
-					progress.End()
-				} else {
+
+				if fix == nil {
 					b.notifier.SendShowMessage(sglsp.MTError, "Oh snap! 😔 The fix did not remediate the issue and was not applied.")
 					progress.End()
+					return nil
 				}
 
-				return edit
+				actionCommandMap, err := b.autofixFeedbackActions(fix.FixId)
+				successMessage := "Congratulations! 🎉 You’ve just fixed this " + issueTitle(issue) + " issue. Was this fix helpful?"
+				if err == nil {
+					b.notifier.Send(snyk.ShowMessageRequest{
+						Message: successMessage,
+						Type:    snyk.Info,
+						Actions: actionCommandMap,
+					})
+				} else {
+					b.notifier.SendShowMessage(sglsp.Info, successMessage)
+				}
+
+				progress.End()
+				return &fix.AutofixEdit
 			}
 		}
 	}
 
 	return editFn
 }
 
+func (b *Bundle) autofixFeedbackActions(fixId string) (*data_structure.OrderedMap[snyk.MessageAction, snyk.Command], error) {
+	createCommandData := func(positive bool) (snyk.Command, error) {
+		commandData := snyk.CommandData{
+			Title:     snyk.CodeSubmitFixFeedback,
+			CommandId: snyk.CodeSubmitFixFeedback,
+			Arguments: []any{fixId, positive},
+		}
+
+		cmd, err := command.CreateFromCommandData(commandData, nil, nil, nil, nil, nil, b.SnykCode)
```

Comment:
This is a clean code / style thingy ;). Calling the factory like this assumes whitebox-knowledge of the factory command creation. I'd advise not to assume how the factory works, thus I'd inject all parameters.

Reply:
It's quite a bad design that Code scanner has to statically depend on all the dependencies of the command factory in the first place. Should a scanner really now about the server? Server is also an application-level dependency. Imagine there's another scanner that wants to reuse the factory. Suddenly it has to depend on Snyk Code Http Client.

Maybe not using a factory would be temporarily better in the current scenario.

P.S. commands are currently in the domain layer, which doesn't make sense for Scanner-specific commands. I believe they should be registered and handled in the infra layer. This could avoid also problems like the above,

Reply:
Happy to discuss these solutions. In the longer run, code scanner should be migrated out of the repository and included as dependency via go.mod. This will need some dependency refactoring. Server is not living in the application layer. If you refer directly to the server, this should change - [this](https://github.com/snyk/snyk-ls/blob/9085f62a3a73a5a6e4e9aaa5ba450fdda6acdc6c/internal/lsp/server.go#L41) should be referenced.

Reply:
You are asking to inject all the parameters, where one of them is reference to server. How do you see this happening now? There're authentication service, issue service dependencies that need to be passed as well.

Reply:
I didn't say that. There's different solutions. 

The problem is that the command is created in the infrastructure layer, not only the command data. The flow is:

```
bundle (infrastructure) -> command factory (ide/command) -> ... -> client
```

The intended flow is that commands are created from command data by the command handler using the command factory and then executed. The command handler has all the necessary dependencies to inject without needing to know how the factory works.

```
client -> command handler -> command factory -> command -> execute -> logic in infrastructure
```

So only CommandData should be used, not the Command. Anyhoo, I'm open to a call to discuss further.

Reply:
Let's address this in a separate PR. I've started work on it in `feat/idea-autofix-feedback/command-data` branch.